File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        ios_base::sync_with_stdio(0);
        vector <char> stringChar;
        int size=0,max_size=0;
        for(char i:s){
            auto pos = find(stringChar.begin(),stringChar.end(),i);
            if(pos != stringChar.end()){
                if(max_size < stringChar.size()){
                    max_size = stringChar.size();
                }
                stringChar.erase(stringChar.begin(),pos+1);
            }
            stringChar.push_back(i);
        }
        if(max_size < stringChar.size()){
            max_size = stringChar.size();
        }
        return max_size;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        values = []
        result = 0

        for i in s:
            if i in values:
                result = max(result,len(values))
                del values[:values.index(i)+1]
            values.append(i)
            
        return max(result,len(values))